src/dashboard/Push/PushIndex.react.js

```diff
@@ -140,15 +140,15 @@ let getPushName = (pushData) => {
       <strong>{title}</strong>
     );
   } else {
-    let payload = pushData[PushConstants.PAYLOAD_FIELD];
+    let payload = pushData[PushConstants.PAYLOAD_FIELD] || '';
     try {
       payload = JSON.parse(payload);
     } catch(e) { }
     if (payload) {
       if (typeof payload.alert === 'string') {
         return payload.alert;
       }
-      return payload.alert ? JSON.stringify(payload.alert) : payload;
+      return payload.alert ? JSON.stringify(payload.alert) : JSON.stringify(payload);
```

Comment:
Cool! One nit is that when the payload is empty string `JSON.stringify('')` results in a double string outputted. We'll only need stringify if payload is an object. Once you've added that check its go to merge. Thanks!